src/main/kotlin/no/nav/familie/ks/sak/barnehagelister/BarnehagelisteVarslingService.kt

```diff
@@ -54,24 +53,58 @@ class BarnehagelisteVarslingService(
     }
 
     // Bruker kommuneNr i stedet for kommunenavn siden det er mindre mulighet for feilskriving
-    private fun finnKommunerSendtInnSisteDøgn(): Set<String> {
+    private fun finnKommunerSendtInnSisteDøgn(): Set<KommuneEllerBydel> {
         val barnSendtInnSisteDøgn =
             barnehageBarnRepository
                 .findAll()
                 .filter { it.endretTidspunkt >= LocalDateTime.now().minusDays(1) }
-        return barnSendtInnSisteDøgn.map { it.kommuneNr }.toSet()
+        return barnSendtInnSisteDøgn.map { KommuneEllerBydel(it.kommuneNr, it.kommuneNavn) }.toSet()
+    }
+
+    private fun hentEnhetTilFylkeMap(): Map<String, Set<String>> {
+        val enhetTilFylkeFil = Thread.currentThread().contextClassLoader.getResource(PATH_ENHETSNR_TIL_FYLKENR_OVERSIKT)
+
+        return objectMapper.readValue(enhetTilFylkeFil, object : TypeReference<Map<String, Set<String>>>() {})
+    }
+
+    private fun hentAnsvarligEnhetForKommuneEllerBydelMap(enhetTilFylkeMap: Map<String, Set<String>>): Map<String, Set<KommuneEllerBydel>> =
+        enhetTilFylkeMap.mapValues { (_, fylkeKoder) ->
+            buildSet {
+                fylkeKoder.forEach { fk ->
+                    val kodeTilNavnIFylke = geografiService.hentBydelEllerKommuneKodeTilNavnFraFylkeNr(fk)
+                    kodeTilNavnIFylke.forEach { (kode, navn) ->
+                        add(KommuneEllerBydel(kode = kode, navn = navn))
+                    }
+                }
+            }
+        }
+
+    private fun finnEnhetForKommuneEllerBydel(
```

Comment:
Hvis `hentAnsvarligEnhetForKommuneEllerBydel` skrives om som foreslått, kan denne funksjonen slettes, for den ender da bare opp med å kalle på den andre funksjonen